framework/src/main/java/org/checkerframework/framework/source/CompositeChecker.java

```diff
@@ -0,0 +1,756 @@
+  protected Set<Class<? extends BaseTypeChecker>> getImmediateSubcheckerClasses() {
+    // This must return a modifiable set because clients modify it.
+    // Most checkers have 1 or fewer subcheckers.
+    LinkedHashSet<Class<? extends BaseTypeChecker>> result =
+        new LinkedHashSet<>(CollectionsPlume.mapCapacity(2));
+    if (shouldResolveReflection()) {
+      result.add(MethodValChecker.class);
+    }
+    return result;
+  }
+
+  /**
+   * Returns whether or not reflection should be resolved.
+   *
+   * @return true if reflection should be resolved
+   */
+  public boolean shouldResolveReflection() {
+    return hasOptionNoSubcheckers("resolveReflection");
+  }
+
+  /** An array containing just {@code BaseTypeChecker.class}. */
+  protected static Class<?>[] baseTypeCheckerClassArray = new Class<?>[] {BaseTypeChecker.class};
+
+  /**
+   * Returns the name of a class related to a given one, by replacing "Checker" or "Subchecker" by
+   * {@code replacement}.
+   *
+   * @param checkerClass the checker class
+   * @param replacement the string to replace "Checker" or "Subchecker" by
+   * @return the name of the related class
+   */
+  @SuppressWarnings("signature") // string manipulation of @ClassGetName string
+  public static @ClassGetName String getRelatedClassName(
+      Class<?> checkerClass, String replacement) {
+    return checkerClass
+        .getName()
+        .replace("Checker", replacement)
+        .replace("Subchecker", replacement);
+  }
+
+  // **********************************************************************
+  // Misc. methods
+  // **********************************************************************
+
+  /** Specify supported lint options for all type-checkers. */
+  @Override
+  public Set<String> getSupportedLintOptions() {
+    Set<String> lintSet = new HashSet<>(super.getSupportedLintOptions());
+    lintSet.add("cast");
+    lintSet.add("cast:redundant");
+    lintSet.add("cast:unsafe");
+
+    for (BaseTypeChecker checker : getSubcheckers()) {
+      lintSet.addAll(checker.getSupportedLintOptions());
+    }
+
+    return Collections.unmodifiableSet(lintSet);
+  }
+
+  /**
+   * Invokes the constructor belonging to the class named by {@code name} having the given parameter
+   * types on the given arguments. Returns {@code null} if the class cannot be found. Otherwise,
+   * throws an exception if there is trouble with the constructor invocation.
+   *
+   * @param <T> the type to which the constructor belongs
+   * @param name the name of the class to which the constructor belongs
+   * @param paramTypes the types of the constructor's parameters
+   * @param args the arguments on which to invoke the constructor
+   * @return the result of the constructor invocation on {@code args}, or null if the class does not
+   *     exist
+   */
+  @SuppressWarnings({"unchecked", "TypeParameterUnusedInFormals"}) // Intentional abuse
+  public static <T> @Nullable T invokeConstructorFor(
```

Comment:
Does `CompositeChecker` need this method? Even if it would like to use this method, I would think that `BaseTypeChecker` would be a better place - a `CompositeChecker` is defined by **not** needing to create a type factory, visitor, etc., which is the main purpose of this method.